abmarl/external/open_spiel_env_wrapper.py

```diff
@@ -0,0 +1,302 @@
+        """
+        return sum([1 for _ in self._learning_agents])
+
+    @property
+    def is_turn_based(self):
+        """
+        The game is turn based if the simulation is wrapped with a TurnBasedManager.
+        """
+        return isinstance(self.sim, TurnBasedManager)
+
+    @property
+    def current_player(self):
+        """
+        The current player is the player that currently provides the action.
+
+        This is used the in the observation part of the TimeStep output. If it
+        is a turn based game, then the current player is the single agent who
+        is providing an action. If it is a simultaneous game, then OpenSpiel does
+        not use this property and the current player is just the first player
+        in the list.
+        """
+        return self._current_player
+
+    @current_player.setter
+    def current_player(self, value):
+        assert value in self._learning_agents, "Current player must be an agent in the simulation."
+        self._current_player = value
+
+    def reset(self, **kwargs):
+        """
+        Reset the simulation.
+        """
+        self._should_reset = False
+        obs = self.sim.reset(**kwargs)
+        self.current_player = next(iter(obs))
+
+        # If it is a turn based sim, then the current player should be the next
+        # player whose turn it will be. But what is current player for an all-step game?
+        observations = {
+            "info_state": self._append_obs(obs),
+            "legal_actions": {
+                agent_id: self.get_legal_actions(agent_id)
+                for agent_id in self._learning_agents
+            },
+            "current_player": self.current_player,
+        }
+
+        return TimeStep(
+            observations=observations,
+            rewards=None,
+            discounts=None,
+            step_type=StepType.FIRST
+        )
+
+    def step(self, action_list, **kwargs):
+        """
+        Step the simulation forward using the reported actions.
+
+        OpenSpiel provides an action list of either (1) the agent whose turn it
+        is in a turn-based game or (2) all the agents in a simultaneous game. The
+        OpenSpielWrapper converts the list of actions to a dictionary before passing
+        it to the underlying simulation.
+
+        OpenSpiel does not support the ability for some agents of a game to finish
+        before others. As such, it may provide actions for agents that are already
+        done. To work with Abmarl, this wrapper removes actions for agents that
+        are already done.
+        """
+        # Actions come in as a list, so we need to convert to a dict before forwarding
+        # to the SimulationManager.
+        if self.is_turn_based:
+            action_dict = {self.current_player: action_list[0]}
+        else:
+            action_dict = {
+                agent_id: action_list[i]
+                for i, agent_id in enumerate(self._learning_agents)
```

Comment:
add assertion for simultaneous simulation that the action list is the same length as the learning agents.